saleor/checkout/utils.py

```diff
@@ -146,10 +146,11 @@ def add_variants_to_checkout(
     channel_listings_by_product_id = {cl.product_id: cl for cl in channel_listings}
 
     # check if variants are published
-    for variant in variants:
-        product_channel_listing = channel_listings_by_product_id[variant.product_id]
-        if not product_channel_listing or not product_channel_listing.is_published:
-            raise ProductNotPublished()
+    for variant, quantity in zip(variants, quantities):
```

Comment:
It was here before but since we are touching this - could you check if we could move this entire validation to `clean_input`? `add_variants_to_checkout` is supposed to only save things and validation should happen earlier.

Reply:
If we move it, we should be able to remove `try..except` block from `clean_input` that catches  `ProductNotPublished` errors.